cmd/bosun/expr/funcs.go

```diff
@@ -384,6 +384,161 @@ var builtins = map[string]parse.Func{
 		F:       V,
 		MapFunc: true,
 	},
+	"aggregate": {
+		Args:   []models.FuncType{models.TypeSeriesSet, models.TypeString, models.TypeString},
+		Return: models.TypeSeriesSet,
+		Tags:   tagFirst,
+		F:      Aggregate,
+	},
+}
+
+// Aggregate combines multiple series matching the specified groups using an aggregator function. If group
+// is empty, all given series are combined, regardless of existing groups.
+// Available aggregator functions include: avg (average), p50 (median), min (minimum) and max (maximum).
+func Aggregate(e *State, T miniprofiler.Timer, series *Results, groups string, aggregator string) (*Results, error) {
+	results := Results{}
+
+	grps := splitGroups(groups)
+	if len(grps) == 0 {
+		// no groups specified, so we merge all group values
+		res, err := aggregate(e, T, series, aggregator)
+		if err != nil {
+			return &results, err
+		}
+		res.Group = opentsdb.TagSet{}
+		results.Results = append(results.Results, res)
+	} else {
```

Comment:
drop the else, return `return &results, nil` above in the if condition, and then the rest of the code does not need to be indented. Same idea under https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow in terms of keeping minimal indentation.

Reply:
Done 👍